### ISSUE TITLE
Default starter throws error when running develop command

### ISSUE THREAD

**Issue by ramiroazar** [gatsbyjs/gatsby#31271]
## Description

Getting the following error after installing `gatsby-starter-default` and running `gatsby develop`.

`Error: Can not convert field 'GatsbyPlugin.options' to InputType`

### Steps to reproduce

1. `gatsby new my-default-starter https://github.com/gatsbyjs/gatsby-starter-default`
2. `cd my-default-starter`
3. `gatsby develop`

### Expected result

The `gatsby develop` command shouldn't throw any errors since it's a fresh installation of the official default starter.

### Actual result

The `gatsby develop` command is throwing the error above.

### Environment

System:
    OS: macOS 11.3.1
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.12 - ~/.nvm/versions/node/v14.16.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 90.0.4430.93
    Firefox: 82.0.2
    Safari: 14.1
  npmPackages:
    gatsby: ^3.4.1 => 3.4.1
    gatsby-plugin-gatsby-cloud: ^2.4.1 => 2.4.1
    gatsby-plugin-image: ^1.4.0 => 1.4.0
    gatsby-plugin-manifest: ^3.4.0 => 3.4.0
    gatsby-plugin-offline: ^4.4.0 => 4.4.0
    gatsby-plugin-react-helmet: ^4.4.0 => 4.4.0
    gatsby-plugin-sharp: ^3.4.1 => 3.4.1
    gatsby-source-filesystem: ^3.4.0 => 3.4.0
    gatsby-transformer-sharp: ^3.4.0 => 3.4.0

**Comment by LekoArts** [comment]
Hi!

Sorry to hear you're running into an issue. I followed your reproduction steps and it works fine for me:

```shell
Development/Playground
➜ gatsby new issue-31271 https://github.com/gatsbyjs/gatsby-starter-default
info Creating new site from git: https://github.com/gatsbyjs/gatsby-starter-default.git

Cloning into 'issue-31271'...
remote: Enumerating objects: 28, done.
remote: Counting objects: 100% (28/28), done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 28 (delta 3), reused 10 (delta 0), pack-reused 0
Receiving objects: 100% (28/28), 612.82 KiB | 1.81 MiB/s, done.
success Created starter directory layout
info Installing packages...

yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "gatsby > eslint-config-react-app@6.0.0" has unmet peer dependency "babel-eslint@^10.0.0".
warning "gatsby > express-graphql@0.9.0" has incorrect peer dependency "graphql@^14.4.1".
warning "gatsby > @typescript-eslint/eslint-plugin > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
warning "gatsby > eslint-plugin-graphql > graphql-config > @endemolshinegroup/cosmiconfig-typescript-loader > ts-node@9.1.1" has unmet peer dependency "typescript@>=2.7".
warning "gatsby-plugin-gatsby-cloud > webpack-assets-manifest@5.0.6" has unmet peer dependency "webpack@^5.2.0".
[4/4] Building fresh packages...
info Initialising git in issue-31271

info Create initial git commit in issue-31271
info
Your new Gatsby site has been successfully bootstrapped. Start developing it by running:

  cd issue-31271
  gatsby develop



Development/Playground took 2m 12s
➜ cd issue-31271

issue-31271 on  main is 📦 v0.1.0 via ⬢ v14.16.1
➜ yarn develop
yarn run v1.22.5
$ gatsby develop
info

There are 6 other flags available that you might be interested in:
- FAST_DEV · Enable all experiments aimed at improving develop server start time
- DEV_SSR · (Umbrella Issue (https://gatsby.dev/dev-ssr-feedback)) · Server Side Render (SSR) pages on full reloads during develop. Helps you detect SSR bugs
and fix them without needing to do full builds.
- PRESERVE_WEBPACK_CACHE · (Umbrella Issue (https://gatsby.dev/cache-clearing-feedback)) · Use webpack's persistent caching and don't delete webpack's cache
when changing gatsby-node.js & gatsby-config.js files.
- PRESERVE_FILE_DOWNLOAD_CACHE · (Umbrella Issue (https://gatsby.dev/cache-clearing-feedback)) · Don't delete the downloaded files cache when changing
gatsby-node.js & gatsby-config.js files.
- PARALLEL_SOURCING · EXPERIMENTAL · (Umbrella Issue (https://gatsby.dev/parallel-sourcing-feedback)) · Run all source plugins at the same time instead of
serially. For sites with multiple source plugins, this can speedup sourcing and transforming considerably.
- FUNCTIONS · EXPERIMENTAL · (Umbrella Issue (https://gatsby.dev/functions-feedback)) · Compile Serverless functions in your Gatsby project and write them to
disk, ready to deploy to Gatsby Cloud

success open and validate gatsby-configs - 0.110s
success load plugins - 1.111s
success onPreInit - 0.027s
success initialize cache - 0.007s
success copy gatsby files - 0.251s
success onPreBootstrap - 0.023s
success createSchemaCustomization - 0.017s
success Checking for changed pages - 0.001s
success source and transform nodes - 0.050s
success building schema - 0.313s
success createPages - 0.001s
success createPagesStatefully - 0.058s
info Total nodes: 43, SitePage nodes: 6 (use --verbose for breakdown)
success Checking for changed pages - 0.001s
success update schema - 0.047s
success write out redirect data - 0.002s
success Build manifest and related icons - 0.059s
success onPostBootstrap - 0.064s
info bootstrap finished - 6.173s
success onPreExtractQueries - 0.001s
success extract queries from components - 1.238s
success write out requires - 0.012s
success run static queries - 0.024s - 2/2 82.47/s
success run page queries - 0.013s - 3/3 235.10/s
⠀
You can now view gatsby-starter-default in the browser.
⠀
  http://localhost:8000/
⠀
View GraphiQL, an in-browser IDE, to explore your site's data and schema
⠀
  http://localhost:8000/___graphql
⠀
Note that the development build is not optimized.
To create a production build, use gatsby build
⠀
success Building development bundle - 9.551s
success onPreExtractQueries - 0.010s
success extract queries from components - 0.072s
success write out requires - 0.011s
success Re-building development bundle - 0.157s
```

Can you [provide a minimal reproduction][reproduction] for this or try it on another machine? If you used npm maybe try yarn (or the other way around)?

[reproduction]: https://gatsby.dev/reproduction

**Comment by ramiroazar** [comment]
Thank you for looking into this, @LekoArts.

I finally got around to testing this on another machine and can confirm it worked fine, even when using the same package manager.

I was hoping to report back with a solution for this particular instance, but I haven't actually been able to fix it yet.

The error is thrown from the following location.

`/.config/yarn/global/node_modules/gatsby-recipes/node_modules/graphql-compose/lib/utils/toInputType.js:100`

I've attempted the following without any luck.

1. Clean global `yarn` cache
2. Reinstall `gatsby-cli`
3. Reinstall `yarn`
4. Use `npm` instead of `yarn`
5. Use `node` version `12.22.1` instead of `14.16.1`

Everything seems to work correctly despite the error.

I'll probably just continue working around this by using `yarn run develop` instead of `gatsby develop` to bypass `gatsby-cli`.
